core/kazoo_call/src/kapps_call_command.erl

```diff
@@ -2446,7 +2511,8 @@ do_collect_digits(#wcc_collect_digits{max_digits=MaxDigits
                     do_collect_digits(Collect#wcc_collect_digits{after_timeout=kz_time:decr_timeout(After, Start)});
                 {'ok', Digit} ->
                     %% DTMF received, collect and start interdigit timeout
-                    Digits =:= <<>>
+                    FlushOnDigit
```

Comment:
We've been moving away from having the right-most arg not returning `boolean()` when using `andalso`/`orelse`.

I would move this to a function `maybe_flush_queue(FlushOnDigit, Digits, Call)` with a definition like

```erlang
maybe_flush_queue('true', <<>>, Call) -> flush(Call);
maybe_flush_queue(_FlushOnDigit, _Digits, _Call) -> 'ok'.
```

Reply:
@jamesaimonetti latest code has been updated for this, please take a look :)